pypsa/pf.py

```diff
@@ -936,7 +936,9 @@ def apply_transformer_types(n: Network) -> None:
 
     # now set calculated values on live transformers
     for attr in ["r", "x", "g", "b", "phase_shift", "s_nom", "tap_side", "tap_ratio"]:
-        n.transformers.loc[trafos_with_types_b, attr] = t[attr]
+        n.transformers.loc[trafos_with_types_b, attr] = t[attr].astype(
+            n.transformers[attr].dtype
+        )
```

Comment:
```suggestion
    attrs = ["r", "x", "g", "b", "phase_shift", "s_nom", "tap_side", "tap_ratio"]
    n.transformers.loc[trafos_with_types_b, attrs] = t[attrs].astype(n.transformers[attrs].dtypes)
```
Could you remove the loop in one go?

Reply:
I did! Also wondering why it was not caught by the global warning rule.